src/login/logind-action.c

```diff
@@ -11,29 +11,115 @@
 #include "logind-dbus.h"
 #include "logind-session-dbus.h"
 #include "process-util.h"
-#include "sleep-config.h"
+#include "sd-messages.h"
 #include "special.h"
 #include "string-table.h"
 #include "terminal-util.h"
 #include "user-util.h"
 
+static const ActionTableItem action_table[_HANDLE_ACTION_MAX] = {
+        [HANDLE_POWEROFF] = {
```

Comment:
I wonder if we need HANDLE_POWEROFF and the other enum entries. Wouldn't it be simpler to pass around `const ActionTableItem*` ?

Reply:
Probably. There's one or two places where it checks for a specific action but that could also be done with calculating the offset of the pointer from action_table I guess.

Reply:
changed the code to use the pointer where possible. I guess the use could be expanded further for someone who doesn't dare to also touch logind-button etc. :-)